### PR TITLE
Put DF_UDF plugin code into the main uber jar.

### PR DESCRIPTION
**Pull request by revans2**
This puts the data frame UDF plugin code into the uber jar. It should now work just like our regular SQL plugin to get DF_UDF functionality.

### CODE REVIEW

**Comment by revans2** [comment]
build

**Review by jlowe — COMMENTED** [review]
Is it necessary to move the code under sql-plugin?  I thought we would just have the aggregator project add the df_udf project as a dependency to pull it into the dist jar, similar to how we treat the shuffle and udf plugin projects.

File: DF_UDF_README.md [hunk 1]
```diff
@@ -15,9 +15,12 @@ commands.
...
 
-To do this include com.nvidia:df_udf_plugin as a dependency for your project and also include it on the 
-classpath for your Apache Spark environment. Then include `com.nvidia.spark.DFUDFPlugin` in the config 
-`spark.sql.extensions`. Now you can implement a UDF in terms of Dataframe operations.
+Accelerator for Apache Spark. As such you will need to select a scala version 2.12 or 2.13 that matches the
+version of Spark you are going to use. You will also need to add it to the
+classpath for your Apache Spark environment. If you plan to not use the GPU accelerated processing then add
+`com.nvidia.spark.DFUDFPlugin` in the config `spark.sql.extensions`. If you do use GPU accelerated processing
+the RAPIDS Plugin will enable this automatically and you don't need to set the `spark.sql.extensions` config
+unless you want to.Now you can implement a UDF in terms of Dataframe operations.
```
Comment (jlowe) [inline DF_UDF_README.md:23]:
```suggestion
unless you want to. Now you can implement a UDF in terms of Dataframe operations.
```

File: DF_UDF_README.md [hunk 2]
```diff
@@ -15,9 +15,12 @@ commands.
 
 ## Setup
 
-To do this include com.nvidia:df_udf_plugin as a dependency for your project and also include it on the 
-classpath for your Apache Spark environment. Then include `com.nvidia.spark.DFUDFPlugin` in the config 
-`spark.sql.extensions`. Now you can implement a UDF in terms of Dataframe operations.
+Accelerator for Apache Spark. As such you will need to select a scala version 2.12 or 2.13 that matches the
```
Comment (abellina) [inline DF_UDF_README.md:18]: It feels something is missing from this sentence. It should at least be "RAPIDS Accelerator for Apache Spark", but my guess is there is a copy and paste error.

**Comment by revans2** [comment]
@jlowe I put it under the SQL plugin because I heard some push back on creating another API jar in addition to a shimed implementation jar. I am fine with doing whatever people want me to do, but we just need to decide what that is. This patch was by far the simplest way to get something working so I went with this.

**Comment by jlowe** [comment]
> This patch was by far the simplest way to get something working so I went with this.

What I'm proposing is even simpler.  Given it's essentially a standalone component, we could leave everything in the existing df_udf project.  We add the df_udf project to the aggregator project so it ends up being part of the dist jar.  The df_udf project would depend on the sql-plugin-api so it can get access to ShimLoader, and we unshim the "front door" classes for df_udf.  That would let us keep the code in a separate project which would make it easier to separate if we ever decide to do that.

I'm fine with it being part of sql-plugin if that's the way you want to go.  I was wondering if you considered the aggregator approach and eliminated it for some reason.

File: DF_UDF_README.md (hunk 2 above)
Comment (revans2) [reply DF_UDF_README.md]: Thanks I totally missed that.

**Comment by revans2** [comment]
@jlowe I thought about it, but I wasn't sure how the shimming would work. I am happy to give it a try and get back to you on it.

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
@jlowe and @abellina please take another look. I tried to do what Jason Suggested, but I ran into some issue with the public APIs not working as I expected. I think I could make it work, but this is already working and I don't want to spend too much more time on it right now.

**Review by jlowe — APPROVED** [review]

**Review by abellina — APPROVED** [review]

**Pull request merged**